scripts/unassign-inactive-issues.mjs

```diff
@@ -0,0 +1,169 @@
+          if (issue.pull_request) {
+            console.log(`Issue #${issue.number} is a pull request`);
+            return true;
+          }
+
+          // Method 2: Check timeline events
+          const timeline = await octokit.rest.issues.listEventsForTimeline({
+            owner,
+            repo,
+            issue_number: issue.number,
+            per_page: 100
+          });
+
+          console.log(`Timeline events for issue #${issue.number}:`, 
+            timeline.data.map(event => ({
+              event: event.event,
+              type: event.event_type,
+              actor: event.actor?.login,
+              created_at: event.created_at
+            }))
+          );
+
+          // Look for various PR-related events
+          const hasPRLink = timeline.data.some(event => 
+            event.event === 'cross-referenced' ||
+            event.event === 'connected' ||
+            event.event === 'referenced' ||
+            (event.source && event.source.type === 'pull_request')
+          );
+
+          if (hasPRLink) {
+            console.log(`Issue #${issue.number} has PR links in timeline`);
+            return true;
+          }
+
+          // Method 3: Check issue body for PR links
+          const prLinkRegex = /(?:close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\s+#\d+/i;
+          if (issue.body && prLinkRegex.test(issue.body)) {
+            console.log(`Issue #${issue.number} has PR references in body`);
+            return true;
+          }
+
+          console.log(`No PR links found for issue #${issue.number}`);
+          return false;
+        } catch (error) {
+          console.error(`Error checking PR links for issue #${issue.number}:`, error);
+          return false;
+        }
+      };
+
+      const lastActivity = new Date(issue.updated_at);
+      const now = new Date();
+      
+      if (now - lastActivity > inactivityPeriodInMinutes * 60 * 1000) {
+        const hasLinkedPR = await checkLinkedPRs();
+        
+        if (hasLinkedPR) {
+          console.log(`Skipping issue #${issue.number} as it has linked PR(s)`);
+          continue;
+        }
+
+        console.log(`Processing inactive issue #${issue.number} with no linked PRs`);
+        
+        try {
+          // Unassign user
+          await octokit.rest.issues.update({
+            owner,
+            repo,
+            issue_number: issue.number,
+            assignees: [],
+          });
+
+          console.log(`Successfully unassigned user from issue #${issue.number}`);
+
+          // Add comment
+          await octokit.rest.issues.createComment({
+            owner,
+            repo,
+            issue_number: issue.number,
+            body: `Automatically unassigning @${assignee.login} due to inactivity. @${assignee.login}, if you're still interested in this issue or already have work in progress, please message us here, and we'll assign you again. Thank you!`,
```

Comment:
@rtibbles suggested we improve this a bit:

_Automatically unassigning @${assignee.login} due to no comments here, or updates on the associated pull request for 1 month._

Reply:
No Problem , I'll Change this

Reply:
Thanks a lot!

Reply:
@SukhvirKooner I'm sorry by this feedback I meant the first part of the sentence. We'd still like to keep _"@${assignee.login}, if you're still interested in this issue or already have work in progress, please message us here, and we'll assign you again. Thank you!"_

So putting together it's

`Automatically unassigning @${assignee.login} due to no comments here, or updates on the associated pull request for 1 month. @${assignee.login}, if you're still interested in this issue or already have work in progress, please message us here, and we'll assign you again. Thank you!`

Reply:
No problem at all! I’ll make the changes to keep the message as you suggested. Thanks for the clarification!